spec/models/book_spec.rb

```diff
@@ -0,0 +1,55 @@
+require "rails_helper"
+
+RSpec.describe Book, type: :model do
+  describe "associations" do
+    it "must belong to shop" do
+      is_expected.to belong_to :shop
+    end
+    it "must belong to category" do
+      is_expected.to belong_to :category
+    end
+    it "has many order_details" do
+      is_expected.to have_many :order_details
+    end
+  end
+
+  describe "validate" do
+    context "title" do
+      it "when nil is invalid" do
+        expect validate_presence_of(:title).with_message I18n.t("activerecord.errors.models.book.attributes.title.blank")
+      end
+    end
+  end
+
+  describe "scope" do
+    let!(:book){FactoryBot.create :book}
```

Comment:
phân biệt giữa let và let! đi em

Reply:
Dạ theo em tìm hiểu thì :
- let : Biến sẽ được tạo lúc chúng ta gọi nó bên trong "it"
- let! : Biến sẽ được tạo ngay trước "it"
Và cả 2 đều cache dữ liệu chỉ trong phạm vi "it"